packages/design-system/src/components/web-components/preactement/define.ts

```diff
@@ -138,6 +145,40 @@ function setupElement<T>(component: ComponentFunction<T>, options: IOptions = {}
   };
 }
 
+function proxyEvents(props, eventNames, CustomElement) {
+  const callbacks = {};
+
+  (eventNames || []).forEach((name) => {
+    const customName = kebabCaseIt(name.replace('on', 'ds'));
+    let existingCallback = () => null;
+
+    // Don't know why `existingCallback` is being defined
+    // Why would a callback already be defined in props?
+    if (props && props[name]) {
+      existingCallback = props[name].bind({});
+    }
+
+    const customCb = (event) => {
```

Comment:
This could also use a comment along the lines of "this is the callback function that will be passed to the preact component and will dispatch a custom event when called."